src/plugins/intel_gpu/src/kernel_selector/kernels/fully_connected/fully_connected_kernel_gemv.cpp

```diff
@@ -0,0 +1,329 @@
+﻿// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "fully_connected_kernel_gemv.h"
+
+#include "common_types.h"
+#include "kernel_selector_utils.h"
+#include "swiglu/swiglu_kernel_base.h"
+
+static constexpr size_t simd = 16;
+
+namespace kernel_selector {
+ParamsKey FullyConnected_GEMV::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableInputWeightsType(WeightsType::INT4);
+    k.EnableInputWeightsType(WeightsType::UINT4);
+    k.EnableInputLayout(DataLayout::bf);
+    k.EnableInputLayout(DataLayout::bfyx);
+    k.EnableOutputLayout(DataLayout::bf);
+    k.EnableOutputLayout(DataLayout::bfyx);
+    k.EnableBiasPerOutput();
+    k.EnableNonBiasTerm();
+    k.EnableTensorOffset();
+    k.EnableDifferentInputWeightsTypes();
+    k.EnableDynamicShapesSupport();
+    k.EnableWeightsCompression();
+    k.EnableBatching();
+    k.EnableBiasPerFeature();
+    k.EnableDifferentTypes();
+    k.EnableSingleBatchOpt();
+
+    return k;
+}
+
+DeviceFeaturesKey FullyConnected_GEMV::get_required_device_features_key(const Params& params) const {
+    auto k = get_common_subgroups_device_features_key(params);
+    k.requires_subgroup_broadcast();
+    k.requires_blocked_read_write();
+    k.requires_blocked_read_write_char();
+    k.requires_blocked_read_write_short();
+
+    return k;
+}
+
+static bool is_weight_vertical(const fully_connected_params& params, size_t output_f) {
```

Comment:
There are quite a lot of helper function copied from bf_tiled kernel, I think we can reuse them

Reply:
yes, this gemv kernel always be called in the second tokens, which need keep the same weight layout with first token to avoid weights reorder.